pandas/core/frame.py

```diff
@@ -4887,11 +4921,16 @@ def sort_index(
         axis = self._get_axis_number(axis)
         labels = self._get_axis(axis)
 
+        # apply key to each level separately and create a new index
```

Comment:
hmm, can we handle this inside `ensure_key_mapped`, e.g. by passing an optional `level` arg?